include/xtensor/xmanipulation.hpp

```diff
@@ -0,0 +1,642 @@
+     */
+    template <class E>
+    inline auto flip(E&& e, std::size_t axis)
+    {
+        using shape_type = xindex_type_t<typename std::decay_t<E>::shape_type>;
+
+        shape_type shape;
+        resize_container(shape, e.shape().size());
+        std::copy(e.shape().cbegin(), e.shape().cend(), shape.begin());
+
+        get_strides_t<shape_type> strides;
+        decltype(auto) old_strides = detail::get_strides(e);
+        resize_container(strides, old_strides.size());
+        std::copy(old_strides.cbegin(), old_strides.cend(), strides.begin());
+
+        strides[axis] *= -1;
+        std::size_t offset = static_cast<std::size_t>(static_cast<std::ptrdiff_t>(e.data_offset()) + old_strides[axis] * (static_cast<std::ptrdiff_t>(e.shape()[axis]) - 1));
+
+        return strided_view(std::forward<E>(e), std::move(shape), std::move(strides), offset);
+    }
+
+    template <std::ptrdiff_t N>
+    struct rot90_impl;
+
+    template <>
+    struct rot90_impl<0>
+    {
+        template <class E>
+        inline auto operator()(E&& e, const std::array<std::size_t, 2>& /*axes*/)
+        {
+            return strided_view(std::forward<E>(e), {});
+        }
+    };
+
+    template <>
+    struct rot90_impl<1>
+    {
+        template <class E>
+        inline auto operator()(E&& e, const std::array<std::size_t, 2>& axes)
+        {
+            xtensor<ptrdiff_t, 1> axes_list = arange<std::ptrdiff_t>(0, std::ptrdiff_t(e.shape().size()));
+            std::swap(axes_list.at(axes[0]), axes_list.at(axes[1]));
+
+            return transpose(flip(std::forward<E>(e), axes[1]), std::move(axes_list));
+        }
+    };
+
+    template <>
+    struct rot90_impl<2>
+    {
+        template <class E>
+        inline auto operator()(E&& e, const std::array<std::size_t, 2>& axes)
+        {
+            return flip(flip(std::forward<E>(e), axes[0]), axes[1]);
+        }
+    };
+
+    template <>
+    struct rot90_impl<3>
+    {
+        template <class E>
+        inline auto operator()(E&& e, const std::array<std::size_t, 2>& axes)
+        {
+            xtensor<ptrdiff_t, 1> axes_list = arange<std::ptrdiff_t>(0, std::ptrdiff_t(e.shape().size()));
+            std::swap(axes_list.at(axes[0]), axes_list.at(axes[1]));
+
+            return flip(transpose(std::forward<E>(e), std::move(axes_list)), axes[1]);
+        }
+    };
+
+    /**
+     * @brief Rotate an array by 90 degrees in the plane specified by axes.
+     * Rotation direction is from the first towards the second axis.
+     *
+     * @param e the input xexpression
+     * @param axes the array is rotated in the plane defined by the axes. Axes must be different.
+     * @tparam N number of times the array is rotated by 90 degrees. Default is 1.
+     *
+     * @return returns a view with the result of the rotation
+     */
+    template <std::ptrdiff_t N = 1, class E>
+    inline auto rot90(E&& e, const std::array<std::size_t, 2>& axes = {0, 1})
```

Comment:
i've added a function called forward_normalize that could be applied here. 
It normalizes the axes (or an axes container) so that you can also use negative axes :)

Reply:
Thanks :) I'll use that